### PR TITLE
React

### LINKED ISSUE #9: Reloading plugin overrides preferences
I updated by PR template URL but when I reload the plugin, it returns to the original.

### LINKED ISSUE #12: Should append a \n and the PR template to innerHTML instead of overwriting

### PR DESCRIPTION
**Pull request by warbrett**
#### What's this PR do?

Converts all html files to React. Changes the pull request button to autofill by left clicking.  
#### Where should the reviewer start?

Start with the src/options.js file this contains nearly all of the changes. After that review the fillPullBody file which has changed how it inserts.
#### How should this be manually tested?

Install the plugin.  Go to a PR, try clicking the plugin button a few times, try changing the text and clicking the button. Then right click on it and view the options page in all of its react glory, click on/change things in there and have your mind blown even more.
#### Any background context you want to provide?

N/A
#### What are the relevant tickets?

closes #9 
closes #12

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,13 @@
     "build": "gulp",
     "serve": "webpack-dev-server"
   },
-  "dependencies": {},
+  "dependencies": {
+    "babel-core": "^5.3.3",
+    "babel-loader": "^5.0.0",
+    "react": "^0.13.3",
+    "react-style": "^0.5.5",
```
Comment (phated) [inline package.json]: Not a huge fan of react-style

File: src/fillPullBody.js [hunk 2]
```diff
@@ -3,8 +3,14 @@
 chrome.runtime.onMessage.addListener(function(request){
   const prBodyElement = document.getElementById('pull_request_body');
   if(request.fillPR && prBodyElement){
-    chrome.storage.sync.get('prTemplate', function(item){
-      prBodyElement.value = item.prTemplate;
+    chrome.storage.sync.get('prTemplate', function({ prTemplate }){
+      let prBody = prBodyElement.value.replace(prTemplate, '');
```
Comment (phated) [inline src/fillPullBody.js]: i think a const should (can?) be used here

File: src/options.js [hunk 3]
```diff
@@ -1,50 +1,103 @@
...
+const style = require('../style/options.js');
 
-function toggleAutoFill(){
-  chrome.storage.sync.set({ autoFill: autoFill.checked });
-  if(autoFill.checked){
-    chrome.tabs.query({ url: 'https://github.com/*/*' }, function(tabs){
-      _.forEach(tabs, function(tab){
-        chrome.tabs.sendMessage(tab.id, { fillPR: true });
-      });
+const EditUrl = React.createClass({
```
Comment (phated) [inline src/options.js]: this should be using ES6 classes. https://facebook.github.io/react/docs/reusable-components.html#es6-classes

``` js
class EditUrl extends React.Component {

}
```

Comment (phated) [inline src/options.js]: Blog post about converting https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html

File: src/options.js [hunk 4]
```diff
@@ -1,50 +1,103 @@
...
+  },
+  handleUrlChange: function(){
+    this.setState({ deltaUrl: event.target.value });
+  },
+  handleEdit: function(){
+    this.setState({ displayUrl: 'none' });
+    this.setState({ editUrl: 'block' });
+  },
+  handleSubmit: function(){
+    this.setState({ displayUrl: 'block' });
```
Comment (phated) [inline src/options.js]: setState can take an object with multiple keys, no reason to call it multiple times

File: src/options.js [hunk 5]
```diff
@@ -1,50 +1,103 @@
...
+          chrome.tabs.sendMessage(tab.id, { fillPR: true });
+        });
+      });
+    }
+  },
+  render: function(){
+    return (
+      <section styles={[style.options_list]}>
+        <h2>Auto-fill:</h2>
+        <input type={"checkbox"} checked={this.state.autoFill} onChange={this.toggle}>Enable auto-fill</input>
```
Comment (phated) [inline src/options.js]: don't use `{ }` if you are using a static value

Comment (phated) [inline src/options.js]: destructure from state

**Comment by alysonz** [comment]
Fixed.

**Comment by alysonz** [comment]
Updated.

**Pull request merged**